### ISSUE TITLE
Remove wordstem_ngrams()

### ISSUE THREAD

**Issue by koheiw** [quanteda/quanteda#1872]
Users should stem tokens before forming ngrams, so we do not need `wordstem_ngrams()` anymore.

https://github.com/quanteda/quanteda/blob/1d515f2d379647a873ed9b2c8c98504165c3bdb0/R/wordstem.R#L29-L48

**Comment by kbenoit** [comment]
True, and once upon a time this is what we did, but then in response to #127 we added this. What harm does it do?

More concerning to me is #1602 and the fact that the convenience methods such as `stem = TRUE` hard-wire the wordstem options. Those are the ones I think should get removal priority.

**Comment by koheiw** [comment]
I think there was not many functions on tokens when #127 was posted. Removal of `wordstem_ngrams()` is along with our recommendation for user to do whatever possible on tokens, before forming a DFM.

**Comment by kbenoit** [comment]
Right now, `wordstem_ngrams()` is purely internal, and is called when a `tokens_wordstem()` is called on a ngram > 2 tokens, or when `dfm_wordstem()` is called on an ngram > 2 dfm. Removing that function will change functionality, but not the API. The reason we added it is that otherwise, wordstem functions on ngrams would just stem the last word, such as:
```r
> char_wordstem("stemming_stemmed")
[1] "stemming_stem"
```
What we want instead (and what now just works) is:
```r
> tokens("stemming stemmed") %>%
+     tokens_ngrams(n = 2) %>%
+     tokens_wordstem()
tokens from 1 document.
text1 :
[1] "stem_stem"
```

If we remove this functionality, we would probably then want to check whether someone was trying to stem an ngram and warn them.

Why offer stemming of ngrams? Basically because some dfm objects are huge and time-consuming to form, so if there were a large dfm object that a user wanted to try in different ways including after stemming, they might want to stem a pre-existing dfm.

But overall, if we remove the internal detection to do smart stemming, we would be making things harder for users, without making our API any simpler (or indeed, different at all). For that reason I'm in favour of keeping it.

**Comment by kbenoit** [comment]
I'm voting to close this - any reason why we would not want correct stemming of each element of an ngram?

**Comment by koheiw** [comment]
This is a redundant functionality because users can achieve the same if they combine other functions properly: apply `tokens_stem()` before `tokens_ngram()`.

**Comment by kbenoit** [comment]
But what if the tokens are inherited, say from another tokeniser, or compounded following some detection of collocations? It's more accurate to say that it's alternative functionality for most use cases, rather than redundant, since it also covers additional use cases.

And since it's not making the documentation index look any messier, why delete it?

**Comment by koheiw** [comment]
I agree that there are such tokens in foreign objects, so it is fine that such a function exists, but not in the core quanteda. The best place for these functions are the quanteda.extra package!
